### PR TITLE
Add custom loss and metric to simply interactive training

### PR DESCRIPTION
**Pull request by anwai98**
Simplifying several aspects, i.e. loss, metric, unified learning rate, scheduler, layer decay and gradient clipping (several engineered components simplified).

I'll leave this as a draft and will merge this once my current training shows some promise!

### CODE REVIEW

**Comment by anwai98** [comment]
I'll go ahead and merge this. This one looks solid!
